pkg/apis/pipeline/v1/task_validation.go

```diff
@@ -274,6 +274,10 @@ func ValidateParameterTypes(ctx context.Context, params []ParamSpec) (errs *apis
 
 // ValidateType checks that the type of a ParamSpec is allowed and its default value matches that type
 func (p ParamSpec) ValidateType() *apis.FieldError {
+	// For resources not call SetDefaults(), empty type should be valid.
```

Comment:
NIT - suggested comment

> A resource may have been created before type was available in `paramSpec` and thus have no type set

Reply:
sure!

Reply:
couldn't we call SetDefaults() before calling validation instead of just skipping validation?

Reply:
> couldn't we call SetDefaults() before calling validation instead of just skipping validation?

yeah I would prefer this way!

Reply:
(or at least in this function, we should assume no type == defaultType)